orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.groovy

```diff
@@ -147,13 +145,15 @@ class ContextParameterProcessor {
         it.type in ['deploy', 'createServerGroup', 'cloneServerGroup', 'rollingPush'] && it.status == ExecutionStatus.SUCCEEDED
       }.each { deployStage ->
         if (deployStage.context.'deploy.server.groups') {
-          Map deployDetails = [
+          def deployDetails = [
             account    : deployStage.context.account,
             capacity   : deployStage.context.capacity,
             parentStage: deployStage.parentStageId,
             region     : deployStage.context.region ?: deployStage.context.availabilityZones.keySet().first(),
           ]
-          deployDetails.putAll(context.execution?.context?.deploymentDetails?.find { it.region == deployDetails.region } ?: [:])
+          deployDetails.putAll(deployStage.context.deploymentDetails?.find {
+            it.region == deployDetails.region
+          } ?: [:])
```

Comment:
Use the deploy stage's context to get `deploymentDetails` as it will chain back to the relevant find image/bake.